controller_manager/src/controller_manager.cpp

```diff
@@ -840,6 +918,12 @@ ControllerManager::update()
     }
   }
 
+  hw_ret = resource_manager_->write_all_resources();
+
+  if (hw_ret != hardware_interface::return_type::OK) {
```

Comment:
You should have this check twice, the first time right after reading. Right now, you can have a read with a failures, but still update the controllers and write their results.

Reply:
you are right!